### DISCUSSION TITLE
Support for Zigbee Aqara E1 double key wireless switch (WXKG17LM)

### DISCUSSION THREAD

**Discussion by suppeee** [arendst/Tasmota#20611] [Support request]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._

Hi there,
i have a problem with using the Aqara E1 double key wireless switch (WXKG17LM) with tasmota. 

I paired the E1 successfully with the tasmota. Console "zbinfo"-command gives the information:
SENSOR = {"ZbInfo":{"0x0A0C":{"Device":"0x0A0C","IEEEAddr":"0x54EF44100090DB81","ModelId":"lumi.remote.acn004","Manufacturer":"LUMI","Endpoints":[1,2,3],"Config":[],"Reachable":false,"BatteryPercentage":100,"BatteryLastSeenEpoch":1706442305,"LastSeen":26394,"LastSeenEpoch":1706447888,"LinkQuality":123}}}

However pressing a button gives no information on the console, also with the more debug option.
On the tasmota web-startpage the device is shown, but the time is greyed-out (as if the device is offline).
Other sensors (also aqara) work without flaws.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x ] Device used (e.g., Sonoff Basic): Eachen eWeLink ZbBridge Pro Ethernet Zigbee Gateway (ZB-GW03-V1.2)_____
- [ x] Tasmota binary firmware version number used: [Tasmota 13.2.0 by Theo Arends](https://bit.ly/tasmota), Zigbee: EZSP v6.7.9.0_____
  - [ x] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
- [ ] 
```lua
  Configuration output here:
Grp 0, Cmd 'BACKLOG', Idx 1, Len 26, Pld -99, Data 'Template; Module; GPIO 255'
04:54:37.413 CMD: Grp 0, Cmd 'TEMPLATE', Idx 1, Len 0, Pld -99, Data ''
04:54:37.417 RSL: RESULT = {"NAME":"ZB-GW03-V1.2","GPIO":[0,0,3552,0,3584,0,0,0,5793,5792,320,544,5536,0,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,608,640,32,0,0,0,0,0],"FLAG":0,"BASE":1}
04:54:37.647 CMD: Grp 0, Cmd 'MODULE', Idx 1, Len 0, Pld -99, Data ''
04:54:37.649 RSL: RESULT = {"Module":{"0":"ZB-GW03-V1.2"}}
04:54:37.897 CMD: Grp 0, Cmd 'GPIO', Idx 1, Len 3, Pld 255, Data '255'
04:54:37.904 RSL: RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"3552":"Zigbee Tx"},"GPIO3":{"0":"None"},"GPIO4":{"3584":"Zigbee Rx"},"GPIO5":{"0":"None"},"GPIO6":{"0":"None"},"GPIO7":{"0":"None"},"GPIO8":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO11":{"0":"None"},"GPIO12":{"5793":"Zigbee Rst2"},"GPIO13":{"5792":"Zigbee Rst1"},"GPIO14":{"320":"Led_i1"},"GPIO15":{"544":"LedLink"},"GPIO16":{"5536":"ETH POWER"},"GPIO17":{"0":"None"},"GPIO18":{"5600":"ETH MDIO"},"GPIO19":{"0":"None"},"GPIO20":{"0":"None"},"GPIO21":{"0":"None"},"GPIO22":{"0":"None"},"GPIO23":{"5568":"ETH MDC"},"GPIO24":{"0":"None"},"GPIO25":{"0":"None"},"GPIO26":{"0":"None"},"GPIO27":{"0":"None"},"GPIO32":{"608":"I2C SCL1"},"GPIO33":{"640":"I2C SDA1"},"GPIO34":{"32":"Button1"},"GPIO35":{"0":"None"},"GPIO36":{"0":"None"},"GPIO37":{"0":"None"},"GPIO38":{"0":"None"},"GPIO39":{"0":"None"}}


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
04:55:31.936 RSL: STATUS = {"Status":{"Module":0,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"tasmota_C25A74","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
04:55:31.942 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"","RestartReason":"Software reset CPU","Uptime":"0T06:08:32","StartupUTC":"2024-01-28T14:46:59","Sleep":50,"CfgHolder":4617,"BootCount":40,"BCResetTime":"2023-10-19T09:41:15","SaveCount":83}}
04:55:31.946 RSL: STATUS2 = {"StatusFWR":{"Version":"13.2.0(tasmota)","BuildDateTime":"2023-11-17T21:20:43","Core":"2_0_14","SDK":"4.4.6.231011","CpuFrequency":240,"Hardware":"ESP32-D0WD-V3 v3.1","CR":"312/699"}}
04:55:31.951 RSL: STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":3,"MqttLog":0,"SysLog":3,"LogHost":"","LogPort":514,"SSId":["",""],"TelePeriod":300,"Resolution":"55C180C0","SetOption":["00000001","2805C80001000600003C5A0A192800000000","000000A0","00004000","00004000","00000000"]}}
04:55:31.958 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":1480,"Free":1399,"Heap":153,"StackLowMark":4,"PsrMax":0,"PsrFree":0,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"1640EF","FlashFrequency":40,"FlashMode":"DIO","Features":["00000809","9F9AC7CF","00548001","000000CF","010013C0","C0000981","40024080","00200000","5400002C","00000080"],"Drivers":"1,2,3,4,5,7,8,9,10,11,12,16,20,21,23,24,26,27,29,35,38,41,50,52,62,68,82","Sensors":"1,2,3,5,6,127","I2CDriver":"7"}}
04:55:31.964 RSL: STATUS5 = {"StatusNET":{"Hostname":"tasmota-C25A74-6772","IPAddress":"0.0.0.0","Gateway":"192.168.178.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.178.1","DNSServer2":"2a02:2761:207c:c300:464e:6dff:fe33:8b35","Mac":"24:DC:C3:C2:5A:74","IP6Global":"","IP6Local":"","Ethernet":{"Hostname":"tasmota-C25A74-6772-eth","IPAddress":"192.168.178.30","Gateway":"192.168.178.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.178.1","DNSServer2":"2a02:2761:207c:c300:464e:6dff:fe33:8b35","Mac":"24:DC:C3:C2:5A:77","IP6Global":"2a02:2761:207c:c300:26dc:c3ff:fec2:5a77","IP6Local":"fe80::26dc:c3ff:fec2:5a77"},"Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
04:55:31.972 RSL: STATUS7 = {"StatusTIM":{"UTC":"2024-01-28T20:55:31","Local":"2024-01-29T04:55:31","StartDST":"2024-03-31T02:00:00","EndDST":"2024-10-27T03:00:00","Timezone":"+08:00","Sunrise":"15:24","Sunset":"00:41"}}
04:55:31.978 RSL: STATUS10 = {"StatusSNS":{"Time":"2024-01-29T04:55:31"}}
04:55:31.983 RSL: STATUS11 = {"StatusSTS":{"Time":"2024-01-29T04:55:31","Uptime":"0T06:08:32","UptimeSec":22112,"Heap":160,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":5,"Objects":74}}}


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_
pairing the aqara E1 with tasmota, press a button of the E1, nothing happens on the console (especially no zigbee event) 

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
one should see a zigbee event showing a button is pressed

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

On zigbee2mqtt-webpage it says: "In order for this device to work (fully), at least the following firmware is required on your adapter", see link https://www.zigbee2mqtt.io/devices/WXKG17LM.html

on zha-device-handlers it says, a specific "manufacturer_id" would be necessary (manufacturer_id_override = 0x115F", see link: https://github.com/zigpy/zha-device-handlers/issues/1794

I think the problem cannot be solved by a zigbee device plugin - maybe the zigbee-firmware has to be modified?

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by suppeee** [comment]
i tried a little bit more, but yet no solution:

as the information on the z2m said, a firmware update would be necessary i updated the firmware from EZSP 6.7.9.0 to [ncp-uart-sw_6.10.9_115200.ota] found at https://github.com/xsp1989/zigbeeFirmware/blob/master/firmware/ZigbeeBridge_SM-011-signed/ncp-uart-sw_6.10.9_115200.ota.

However my bridge refused to work with this firmware file (error shown on tasmota console when starting zigbee, so no zigbee working at all).

A downgrade to https://github.com/xsp1989/zigbeeFirmware/blob/master/firmware/ZigbeeBridge_SM-011-signed/ncp-uart-sw_6.7.10_115200.ota made the bridge/zigbee work again, nevertheless the switch still does not work. 

SENSOR = {"ZbInfo":{"0x04BC":{"Device":"0x04BC","IEEEAddr":"0x54EF44100090DB81","ModelId":"lumi.remote.acn004","Manufacturer":"LUMI","Endpoints":[1,2,3],"Config":[],"Reachable":true,"BatteryPercentage":42,"BatteryLastSeenEpoch":1706693505,"LastSeen":779,"LastSeenEpoch":1706693515,"LinkQuality":123}}}

any ideas? maybe xsp1989 can give a hint?

**Reply by Sentry03** [reply]
Hi. I'm a new comer and really don't know anything and I'm browsing the web cause I have the same device as yours and it doesn't work as intended. 
![image](https://github.com/arendst/Tasmota/assets/155286392/92f5d25f-1d38-4e00-bc77-1b6a96fb0a19)

Can you help me with this? Why are there no controls available? Thanks
